iot-hub/Samples/service/DigitalTwinClientSamples/Thermostat/ThermostatSample.cs

```diff
@@ -0,0 +1,136 @@
+        {
+            _logger.LogDebug($"Get the {_digitalTwinId} digital twin.");
+
+            var getDigitalTwinResponse = await _digitalTwinClient.GetDigitalTwinAsync<T>(_digitalTwinId);
+            var thermostatTwin = getDigitalTwinResponse.Body;
+            _logger.LogDebug($"{_digitalTwinId} twin: \n{JsonConvert.SerializeObject(thermostatTwin, Formatting.Indented)}");
+
+            return thermostatTwin;
+        }
+
+        private async Task UpdateTargetTemperatureProperty()
+        {
+            const string targetTemperaturePropertyName = "targetTemperature";
+            var updateOperation = new UpdateOperationsUtility();
+
+            // Choose a random value to assign to the targetTemperature property
+            int desiredTargetTemperature = Random.Next(0, 100);
+
+            // First let's take a look at when the property was updated and what was it set to.
+            var getDigitalTwinResponse = await _digitalTwinClient.GetDigitalTwinAsync<ThermostatTwin>(_digitalTwinId);
+            WritableProperty currentTargetTemperature = getDigitalTwinResponse.Body.Metadata.TargetTemperature;
+            if (currentTargetTemperature != null)
+            {
+                var targetTemperatureDesiredLastUpdateTime = getDigitalTwinResponse.Body.Metadata.TargetTemperature.LastUpdateTime;
+                _logger.LogDebug($"The property {targetTemperaturePropertyName} was last updated on {targetTemperatureDesiredLastUpdateTime.ToLocalTime()} `" +
+                    $" with a value of {getDigitalTwinResponse.Body.Metadata.TargetTemperature.DesiredValue}.");
+
+                // The property path to be replaced should be prepended with a '/'
+                updateOperation.AppendReplaceOp($"/{targetTemperaturePropertyName}", desiredTargetTemperature);
+            }
+            else
+            {
+                _logger.LogDebug($"The property {targetTemperaturePropertyName} was never set on the ${_digitalTwinId} digital twin.");
+
+                // The property path to be added should be prepended with a '/'
+                updateOperation.AppendAddOp($"/{targetTemperaturePropertyName}", desiredTargetTemperature);
+            }
+
+            _logger.LogDebug($"Update the {targetTemperaturePropertyName} property on the {_digitalTwinId} digital twin to {desiredTargetTemperature}.");
+            HttpOperationHeaderResponse<DigitalTwinUpdateHeaders> updateDigitalTwinResponse = await _digitalTwinClient.UpdateDigitalTwinAsync(_digitalTwinId, updateOperation.Serialize());
+
+            _logger.LogDebug($"Update {_digitalTwinId} digital twin response: {updateDigitalTwinResponse.Response.StatusCode}.");
+        }
+
+        private async Task UpdateCurrentTemperatureProperty()
+        {
+            // Choose a random value to assign to the currentTemperature property
+            int currentTemperature = Random.Next(0, 100);
+
+            const string currentTemperaturePropertyName = "currentTemperature";
```

Comment:
We should call out explicitly that this property doesn't exist on model that this device has "registered" with

Reply:
Just to refresh my memory, why did we decide against doing these operations on the root level "targetTemperature" directly?

Reply:
I'll add a comment to explicitly call out that this property does not exist on the model registered with this device.

Regarding your question, I don't think we discussed why we can't do these operations on targetTemperature. We can. But I feel like I don't want to remove targetTemperature property because it's in the model. So if the user wants to run this sample again, against the same device id, that targetTemperature won't exist anymore because it's been deleted due to the previous run of the sample.

Reply:
Added a comment to the RunSampleAsync to explicitly make the distinction about currentTemperature